packages/serverless/src/awslambda.ts

```diff
@@ -243,16 +243,22 @@ export function wrapHandler<TEvent, TResult>(
       }, timeoutWarningDelay);
     }
 
-    // Applying `sentry-trace` to context
-    let traceparentData;
+    // Extract tracing data from headers
+    let traceparentData, tracestateData;
     const eventWithHeaders = event as { headers?: { [key: string]: string } };
-    if (eventWithHeaders.headers && isString(eventWithHeaders.headers['sentry-trace'])) {
-      traceparentData = extractTraceparentData(eventWithHeaders.headers['sentry-trace'] as string);
+
+    if (eventWithHeaders.headers?.['sentry-trace']) {
+      traceparentData = extractSentrytraceData(eventWithHeaders.headers['sentry-trace'] as string);
+    }
+    if (eventWithHeaders.headers?.tracestate) {
+      tracestateData = extractTracestateData(eventWithHeaders.headers.tracestate as string);
     }
+
     const transaction = startTransaction({
       name: context.functionName,
       op: 'awslambda.handler',
       ...traceparentData,
+      ...(tracestateData && { metadata: { tracestate: tracestateData } }),
```

Comment:
In node express middleware we don't do it conditionally like here. Is there any specific reason why? Or just missed it?

Reply:
A late change in the serverless implementation which didn't get ported back to express. In the end it doesn't matter (the `Transaction` constructor does the same thing with `metadata: {tracestate: undefined}` as with no `metadata` entry), but you're right that it would be good to be consistent. Thanks for catching it. I'll fix the express one.